include/swift/AST/DiagnosticsSema.def

```diff
@@ -1166,6 +1166,8 @@ NOTE(unwrap_iuo_initializer,none,
 NOTE(unwrap_with_guard,none,
      "short-circuit using 'guard' to exit this function early "
      "if the optional value contains 'nil'", ())
+NOTE(perform_optional_chain_on_closure, none,
+     "use optional chaining to call this closure", ())
```

Comment:
Does this wording sound clear enough? Let me know if you have any suggestions.

Reply:
I think this wording is good! "Optional chaining" is the user-facing term and is easily searchable.

Reply:
Great! Another variation I thought of was "use optional chaining to call this closure _when the optional value is not nil_" and "use optional chaining to call this closure _when the optional closure is not nil_" It seems to match what the other similar diagnostics are saying, but I wasn't satisfied with using "optional value" and "optional closure" to refer to it, so I opted to drop that part and just keep the initial bit. Do you think either of those makes sense to keep? (or do you have suggestions for another phrase I could use that might be clearer?). Because I suppose it's not completely clear _when_ this closure will be called (i.e. when it's not nil).

Reply:
`use optional chaining to call this value of function type when optional is non-'nil'` might be better because  it's technically not known whether it's actually closure or not, could could be partially applied method for example:

```
struct S {
  func test() {
  }
}

func test(s: S?) {
  let fn = s?.test
  fn?()
}
```

Reply:
Thank you, that phrasing does sound better to me! I have also added your example as a test case. Let me know what you think.

Reply:
Looks good, thank you!